lib/salus/scanners/report_go_dep.rb

```diff
@@ -7,38 +7,131 @@
 module Salus::Scanners
   class ReportGoDep < Base
     def run
-      unless should_run?
+      if @repository.go_sum_present?
+        record_dep_from_go_sum
+      elsif @repository.go_mod_present?
+        record_dep_from_go_mod
+      elsif @repository.dep_lock_present?
+        record_dep_from_go_lock_package
+      else
         raise(
           InvalidScannerInvocationError,
-          'Cannot report on Go dependencies without a Gopkg.lock file.'
+          'Cannot report on Go dependencies without a Gopkg.lock, go.mod, or go.sum file'
         )
       end
+    end
+
+    def record_dep_from_go_sum
+      go_sum_path = "#{@repository.path_to_repo}/go.sum"
+      dep_list = []
+
+      File.foreach(go_sum_path).each("=\n") do |line|
+        next if line.empty?
+
+        line_info = get_line_info(line)
+
+        dep_list.append(
+          {
+            "fullDependency" => line,
+            "name" => line_info[0],
+            "version" => line_info[1]
+          }
+        )
+      end
+      # Note references are hashes meant for packages in Gopkg.lock files
+      dep_list.each do |dependency|
+        record_dep_package(
+          name: dependency['name'],
+          reference: "N/A for go.mod/go.sum dependencies",
+          version_tag: dependency['version'],
+          dependency_file: "go.sum",
+          type: "go_sum"
+        )
+      end
+    end
+
+    def record_dep_from_go_mod
+      go_mod_path = "#{@repository.path_to_repo}/go.mod"
+      dep_list = []
+      parse_line = false
+
+      File.foreach(go_mod_path).each("\n") do |line|
+        next if line.empty?
+
+        if line.include? "require ("
```

Comment:
It would be more precise to add a check to eliminate cases like `XYZrequire ("

Also, there could be spaces between `require` and `(`. So it would be best to use a regular expression for the check.

Reply:
Moreover, what if `require (` is just part of a comment?

Reply:
Good point, I'll strip each line first and switch this check out with a regex like `/^require\s{0,} \(/`.

I think that will cover cases like with whitespaces preceding or between require and (. Cases like `XYZrequire (` and comments should be covered too as they start with '#' or similar.